ultravox/data/datasets.py

```diff
@@ -947,6 +949,67 @@ def _get_sample(self, row) -> VoiceSample:
         )
 
 
+class GenericVoiceDataset(VoiceDataset):
+    def __init__(
+        self, args: VoiceDatasetArgs, config: config_base.DataDictConfig
+    ) -> None:
+        super().__init__(args)
+
+        dataset = datasets.concatenate_datasets(
+            [
+                self._load_audio_dataset(
+                    config.path,
+                    name=config.name,
+                    split=s,
+                    streaming=config.streaming,
+                    shuffle=False,
+                )
+                for s in config.splits
+            ]
+        )
+
+        if self._args.shuffle:
+            dataset = dataset.shuffle(seed=self._args.shuffle_seed)
+
+        if config.num_samples:
+            dataset = dataset.select(range(config.num_samples))
+
+        self.user_template = config.user_template
+        self.assistant_template = config.assistant_template
+        self.transcript_template = config.transcript_template
+
+        self._init_dataset(dataset)
+
+    def _get_sample(self, row) -> VoiceSample:
+        try:
+            user_content = jinja2.Template(
+                self.user_template, undefined=jinja2.StrictUndefined
+            ).render(**row, text_proc=text_proc)
+            assistant_content = jinja2.Template(
+                self.assistant_template, undefined=jinja2.StrictUndefined
+            ).render(**row, text_proc=text_proc)
+            transcript = jinja2.Template(
+                self.transcript_template, undefined=jinja2.StrictUndefined
+            ).render(**row, text_proc=text_proc)
+        except jinja2.TemplateError as e:
+            print(f"Error rendering template: {e}")
+            print(f"user_template: {self.user_template}")
+            print(f"assistant_template: {self.assistant_template}")
+            print(f"transcript_template: {self.transcript_template}")
+            print(f"sample keys: {list(row.keys())}")
+            raise ValueError(
+                f"Template rendering failed. Make sure all keys in the template exist in the sample."
+            ) from e
+
+        self._make_sample(
```

Comment:
Should this return the result of _make_sample? I don't understand why this has `_make_sample` and also `_get_transcribe_sample`.

Reply:
Thanks for catching this. This may explain the unstable results I have got so far.